### ISSUE TITLE
Pipeline parameters not enabled error

### ISSUE THREAD

**Issue by davykinger** [MicrosoftDocs/azure-devops-docs#6588]
We get "Pipeline parameters not enabled" error when pressing 'Run' on the build pipeline. This is after adding our parameter values to the top of our yaml file. Anyone know why that is?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 6724abea-bbdc-bf66-ed5e-3214fa6c3e66
* Version Independent ID: 4f8dab21-3f0e-da32-cc0e-1d85c13c0065
* Content: [Templates - Azure Pipelines](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops#feedback)
* Content Source: [docs/pipelines/process/templates.md](https://github.com/MicrosoftDocs/vsts-docs/blob/master/docs/pipelines/process/templates.md)
* Product: **devops**
* Technology: **devops-cicd**
* GitHub Login: @juliakm
* Microsoft Alias: **jukullam**

**Comment by snickler** [comment]
From what I discovered, you can only use parameters in YAML files if they are within templates. You're better off defining variables in the pipeline in the form of 'Parameters.name' if you copied the YAML output from a classic pipeline.

**Comment by msebolt** [comment]
@davykinger Thank you for the feedback.

Unfortunately, since this is a documentation forum our support options are limited.

Please start an issue [here](https://developercommunity.visualstudio.com/topics/Azure+DevOps.html) for feedback from the product team.

**Comment by rdbartram** [comment]
I am having this same issue. Docs say its supported here...but it isn't https://docs.microsoft.com/en-us/azure/devops/pipelines/process/runtime-parameters?view=azure-devops

**Comment by rdbartram** [comment]
i've checked the preview features and I have almost everything turned on for the Org except Experimental Themes and Full access to Azure Pipelines for Stakeholders

**Comment by msebolt** [comment]
Please refer to [this open issue](https://developercommunity.visualstudio.com/content/problem/872011/passing-variables-in-azure-pipelines.html) with the product team currently under investigation.

**Comment by rdbartram** [comment]
@msebolt Unless I am grossly misunderstanding that issue you sent...this isn't the same issue. My problem is that I can't define paramters within the parent azure-pipielines.yml file and have these basically as "Settable at Queue Time". I'm trying to following the example here https://docs.microsoft.com/en-us/azure/devops/pipelines/process/runtime-parameters?view=azure-devops#use-parameters-in-pipelines

**Comment by rdbartram** [comment]
You may be right that this may be a technical issue that needs to be resolved...but the docs need to reflect that this currently isn't supported uniformly across all tenants yet or whatever.

**Comment by msebolt** [comment]
@rdbartram If this is a new issue I'll need to assign this to the author for further review.

Thank you for bringing this to our attention.

**Comment by jason-ha** [comment]
I found the same docs today while looking for a nice way to do a loop over several steps. I found two odd results (1) triggers seem to be ignored and pipeline is run automatically when I push (only since adding parameters) and (2) attempting manual run anyway there is error "Pipeline parameters not enabled".
I have played further and found that the bad trigger was actually generated because there were compile errors for my pipeline. Those errors happen to be from trying to use `${{ parameters.foo }}` as recently documented. When there were no errors, there was no entry in the pipeline history. I then was able to add a sample parameters block to my pipeline so long as I didn't try to use any of it. This presumable compiles fine because I received no error. When I tried to manually run however I did not see a `Run pipeline` result similar to the documented example. The single parameter I added was nowhere to be found.
Perhaps the docs were published ahead of full rollout of support for this?
